### ISSUE TITLE
Feature request: pre-commit or pre-push only hooks

### ISSUE THREAD

**Issue by ymilki** [pre-commit/pre-commit#255]
I am writing a pre-commit hook that validates the author and committer emails of commits. However, these values can only be guaranteed to queried properly *after* the commit is created.

git does not provide utilities to retrieve this information in the pre-commit stage nor can you retrieve explicit information from the --author flag. (http://stackoverflow.com/questions/9627198/stop-a-git-commit-by-a-specific-author-using-pre-commit-hook)

I would like to ensure that this hook is only run at pre-push time. I can think of two implementations:

* yaml config file can specify pre-push or pre-commit only
* pre-push/pre-commit stage is made available via environmental variable when hook is run.

**Comment by ymilki** [comment]
Turns out, this is partially blocked by #256 since commits can be made without changed files.

**Comment by dnephin** [comment]
I have a use for this as well

**Comment by barrysteyn** [comment]
I would also need something like this

**Comment by asottile** [comment]
I think the way I'd like to see this implemented is a per-hook `stage` option (that defaults to a list of all the stages, currently `['commit', 'push']`).

Probably something like:

```yaml
-   repo: ...
    sha: ...
    hooks:
    -   id: check-added-large-files
        stages: [commit]
    -   id: check-merge-conflict
        stages: [push]
```

Does this seem reasonable?

**Comment by barrysteyn** [comment]
Hi

I like this. But one thing... Can stages mimic the names of the hooks, e.g:

```
-   repo: ...
    sha: ...
    hooks:
    -   id: check-added-large-files
        stages: [commit]
    -   id: check-merge-conflict
        stages: [pre-commit,pre-push]
```

**Comment by barrysteyn** [comment]
@asottile  Actually, after thinking about it some more, ignore my last comment. I like yours much better. stages will be commit or push.

**Comment by barrysteyn** [comment]
This issue is now solved in #271
